src/Storages/DataLakes/IcebergMetadataParser.cpp

```diff
@@ -0,0 +1,301 @@
+
+        const auto metadata_files = MetadataReadHelper::listFiles(configuration, metadata_directory, metadata_file_suffix);
+        if (metadata_files.empty())
+        {
+            throw Exception(
+                ErrorCodes::FILE_DOESNT_EXIST,
+                "The metadata file for Iceberg table with path {} doesn't exist",
+                configuration.url.key);
+        }
+
+        /// Get the latest version of metadata file: v<V>.metadata.json
+        return *std::max_element(metadata_files.begin(), metadata_files.end());
+    }
+
+    /**
+     * In order to find out which data files to read, we need to find the `manifest list`
+     * which corresponds to the latest snapshot. We find it by checking a list of snapshots
+     * in metadata's "snapshots" section.
+     *
+     * Example of metadata.json file.
+     *
+     * {
+     *     "format-version" : 1,
+     *     "table-uuid" : "ca2965ad-aae2-4813-8cf7-2c394e0c10f5",
+     *     "location" : "/iceberg_data/db/table_name",
+     *     "last-updated-ms" : 1680206743150,
+     *     "last-column-id" : 2,
+     *     "schema" : { "type" : "struct", "schema-id" : 0, "fields" : [ {<field1_info>}, {<field2_info>}, ... ] },
+     *     "current-schema-id" : 0,
+     *     "schemas" : [ ],
+     *     ...
+     *     "current-snapshot-id" : 2819310504515118887,
+     *     "refs" : { "main" : { "snapshot-id" : 2819310504515118887, "type" : "branch" } },
+     *     "snapshots" : [ {
+     *       "snapshot-id" : 2819310504515118887,
+     *       "timestamp-ms" : 1680206743150,
+     *       "summary" : {
+     *         "operation" : "append", "spark.app.id" : "local-1680206733239",
+     *         "added-data-files" : "1", "added-records" : "100",
+     *         "added-files-size" : "1070", "changed-partition-count" : "1",
+     *         "total-records" : "100", "total-files-size" : "1070", "total-data-files" : "1", "total-delete-files" : "0",
+     *         "total-position-deletes" : "0", "total-equality-deletes" : "0"
+     *       },
+     *       "manifest-list" : "/iceberg_data/db/table_name/metadata/snap-2819310504515118887-1-c87bfec7-d36c-4075-ad04-600b6b0f2020.avro",
+     *       "schema-id" : 0
+     *     } ],
+     *     "statistics" : [ ],
+     *     "snapshot-log" : [ ... ],
+     *     "metadata-log" : [ ]
+     * }
+     */
+    struct Metadata
+    {
+        int format_version;
+        String manifest_list;
+        Strings manifest_files;
+    };
+    Metadata processMetadataFile(const Configuration & configuration, ContextPtr context)
+    {
+        const auto metadata_file_path = getMetadataFile(configuration);
+        auto buf = MetadataReadHelper::createReadBuffer(metadata_file_path, context, configuration);
+        String json_str;
+        readJSONObjectPossiblyInvalid(json_str, *buf);
+
+        Poco::JSON::Parser parser; /// For some reason base/base/JSON.h can not parse this json file
```

Comment:
as I understand, it failed to parse because of new line symbol before the next value (I got exception about "expected \\", got \n"), e.g.
```
},
"current-schema-id" : 0,
```
This json works fine with `file` table function, but looks like not with json-functions just because it expects json as a single string so I cannot make an example to fail this json as I cannot put this json into json function argument without removing new line symbols so the problem becomes non reproducible.